auth0/resource_auth0_organization.go

```diff
@@ -74,83 +69,32 @@ func newOrganization() *schema.Resource {
 						"assign_membership_on_login": {
 							Type:     schema.TypeBool,
 							Optional: true,
+							Default:  false,
 						},
 					},
 				},
-				Set: hash.StringKey("connection_id"),
```

Comment:
We can't use just the connection_id for the set hashes as that will ignore anything that we update on the `assign_membership_on_login` field. This is mostly the core of the fix, but in order to make it work properly several other things were needed. Please follow comments.